docroot/modules/custom/foia_quarterly_data_report/src/Plugin/Action/QuarterlyReportModerationAction.php

```
@@ -0,0 +1,296 @@
+  /**
+   * Message.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * Validation.
+   *
+   * @var \Drupal\content_moderation\StateTransitionValidationInterface
+   */
+  protected $validation;
+
+  /**
+   * Moderation Info.
+   *
+   * @var \Drupal\content_moderation\ModerationInformationInterface
+   */
+  protected $moderationInfo;
+
+  /**
+   * QuarterlyReportModerationAction constructor.
+   *
+   * @param array $configuration
+   *   Config service.
+   * @param string $plugin_id
+   *   Plugin id service.
+   * @param string $plugin_definition
+   *   Plugin definition service.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   Messenger service.
+   * @param \Drupal\content_moderation\ModerationInformationInterface $moderation_info
+   *   Moderation information service.
+   * @param \Drupal\content_moderation\StateTransitionValidationInterface $validation
+   *   Validation information service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, MessengerInterface $messenger, ModerationInformationInterface $moderation_info, StateTransitionValidationInterface $validation) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->messenger = $messenger;
+    $this->moderationInfo = $moderation_info;
+    $this->validation = $validation;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('messenger'),
+      $container->get('content_moderation.moderation_information'),
+      $container->get('content_moderation.state_transition_validation')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($entity = NULL) {
+    if (is_array($entity)) {
```

Is it correct that this will silently do nothing if $entity is not an array?